tests/Functional/AuditingTest.php

```diff
@@ -620,13 +620,46 @@ public function itWillAuditSync()
 
         $article->auditSync('categories', [$secondCategory->getKey()]);
 
+        $audit = $article->audits()->where('event', 'sync')->latest()->first();
         $no_of_audits_after = Audit::where('auditable_type', Article::class)->count();
         $categoryAfter = $article->categories()->first()->getKey();
 
         $this->assertSame($firstCategory->getKey(), $categoryBefore);
         $this->assertSame($secondCategory->getKey(), $categoryAfter);
         $this->assertNotSame($categoryBefore, $categoryAfter);
         $this->assertGreaterThan($no_of_audits_before, $no_of_audits_after);
+        $this->assertCount(1, $audit->old_values['categories']);
+        $this->assertCount(1, $audit->new_values['categories']);
+    }
+
+    /**
+     * @test
+     * @return void
+     */
+    public function itWillAuditSyncAddAndRemove()
+    {
+        $firstCategory = factory(Category::class)->create();
+        $secondCategory = factory(Category::class)->create();
+        $thirdCategory = factory(Category::class)->create();
+        $article = factory(Article::class)->create();
+
+        $article->categories()->attach($firstCategory);
+
+        $no_of_audits_before = Audit::where('auditable_type', Article::class)->count();
+
+        $article->auditSync('categories', [$secondCategory->getKey(), $thirdCategory->getKey()]);
```

Comment:
are you expecting this to **add** two categories or **set** the whole relation to just these two categories?

Reply:
To **set** it to the two categories. I'm also expecting that "old" contains the first category (i.e. the same as `$article->categories` before the sync), and "new" the second and third (i.e. the same as `$article->categories` after the sync).

Reply:
Makes sense thanks, looks like @erikn69 tweaked the test so it fails now so hopefully is on the case 🙏